lib/runtime/storage/trie.go

```diff
@@ -176,34 +191,40 @@ func (t *TrieState) NextKey(key []byte) []byte {
 	defer t.mtx.RUnlock()
 
 	if currentTx := t.getCurrentTransaction(); currentTx != nil {
-		allEntries := t.state.Entries()
-		maps.Copy(allEntries, currentTx.upserts)
-
-		keys := maps.Keys(allEntries)
-		sort.Strings(keys)
-
-		for _, k := range keys {
-			if k > string(key) && !currentTx.deletes[k] {
-				return []byte(k)
-			}
+		allSortedKeys := append(t.sortedKeys, currentTx.sortedKeys...)
+		// Find key position
```

Comment:
How do you check if the key is not in the `delete` set?

Reply:
We don't need to do that since we are removing them from the `sortedKeys` list when we remove keys from the trie.